### PR TITLE
added check_api_doc.py module in doc/source

### PR DESCRIPTION
**Pull request by alixdamman**
This is to be discussed but it would be nice to have something (even if ad-hoc) to check if something is missing in the file api.rst

### CODE REVIEW

File: doc/source/check_api_doc.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    def add_item(item, api_doc_items):
+        item = item.astext().strip().split()
+        if len(item):
+            if isinstance(item, list):
+                item = item[0]
+            api_doc_items.append(item)
+
+    api_doc_items = []
+    directives.register_directive('autosummary', Autosummary)
+    cleanup_line = lambda line: line.replace(':attr:', '      ').replace('`', ' ')
```
Comment (gdementen) [inline doc/source/check_api_doc.py]: I don't really like lambdas... Yes they save one line, but they are less readable. Not worth it IMO.

**Comment by alixdamman** [comment]
I found weird that users could modify the data on an array without any check on the axes and reversely. 
But I'm aware that implementing properties could impact the general performance of LArray objects. 
I'm OK to move the last 2 commits in another branch (PR).

**Comment by gdementen** [comment]
shouldn't we run that script automatically somewhere during the release process ?

**Comment by gdementen** [comment]
I don't know. What are the pro and con?

**Comment by alixdamman** [comment]
Sounds like Numpy 1.16 is incompatible with the latest release of pytables : https://stackoverflow.com/questions/54210073/pd-read-df-throws-cannot-set-writable-flag-to-true-of-this-array

This makes unittests testing read and write files in HDF format to fail

File: setup.py [hunk 2]
```diff
@@ -15,7 +15,8 @@ def readlocal(fname):
 DESCRIPTION = "N-D labeled arrays in Python"
 LONG_DESCRIPTION = readlocal("README.rst")
 SETUP_REQUIRES = []
-INSTALL_REQUIRES = ['numpy >= 1.13', 'pandas >= 0.13.1']
+# excludes numpy 1.16.* because it is incompatible with pytables 3.4
```
Comment (gdementen) [inline setup.py]: AFAICT, it dates back much further than 3.4. I would write something like "pytables < 3.5 (which is not released yet)"

**Pull request merged**